src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java

```diff
@@ -114,12 +118,24 @@ private Completable doTransfer(T newMessage, SqsQueue<T> newQueue, Optional<Dura
                 .flatMapCompletable((msgId) -> sqsQueue.deleteMessage(receiptId));
     }
 
+    /**
+     * @return a Single that is completed when the ack mode is chosen, or the visibility timeout expires. If it expires,
+     * this will return AckMode.IGNORE
+     */
     public Single<AckMode> getAckMode() {
-        return ackModeSingle;
+        return ackModeSingle
+                .timeout(timeout.getSeconds(), TimeUnit.SECONDS)
```

Comment:
like if I were to call this method more than once I would get a new 'timeout' each time?